File: Editorial/55_Jump_Game/CSharp/1_Backtracking.cs
public class Solution {
    public bool CanJumpFromPosition(int position, int[] nums) {
        if (position == nums.Length - 1) {
            return true;
        }

        int furthestJump = Math.Min(position + nums[position], nums.Length - 1);
        for (int nextPosition = position + 1; nextPosition <= furthestJump;
             nextPosition++) {
            if (CanJumpFromPosition(nextPosition, nums)) {
                return true;
            }
        }

        return false;
    }

    public bool CanJump(int[] nums) {
        return CanJumpFromPosition(0, nums);
    }
}

File: Editorial/55_Jump_Game/TypeScript/1_Backtracking.ts
function canJumpFromPosition(position: number, nums: number[]): boolean {
    if (position == nums.length - 1) {
        return true;
    }
    const furthestJump = Math.min(position + nums[position], nums.length - 1);
    for (
        let nextPosition = position + 1;
        nextPosition <= furthestJump;
        nextPosition++
    ) {
        if (canJumpFromPosition(nextPosition, nums)) {
            return true;
        }
    }
    return false;
}
function canJump(nums: number[]): boolean {
    return canJumpFromPosition(0, nums);
}
